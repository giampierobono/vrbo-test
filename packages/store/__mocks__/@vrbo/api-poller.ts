// tslint:disable-next-line:no-var-requires
const apiPollerStore = require("../../../../__mocks__/@vrbo/api-poller");

module.exports = apiPollerStore;
