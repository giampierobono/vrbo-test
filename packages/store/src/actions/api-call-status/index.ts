export * from "./api-call-status.actions";
