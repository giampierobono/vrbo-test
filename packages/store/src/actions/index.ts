export * from "./api-call-status";
export * from "./poller-config";
export * from "./properties-sort";
export * from "./property";
