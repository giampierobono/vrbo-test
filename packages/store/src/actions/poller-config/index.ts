export * from "./poller-config.actions";
