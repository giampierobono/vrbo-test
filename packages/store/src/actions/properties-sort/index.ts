export * from "./properties-sort.actions";
