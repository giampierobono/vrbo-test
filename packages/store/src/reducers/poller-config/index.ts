export * from "./poller-config.reducer";
