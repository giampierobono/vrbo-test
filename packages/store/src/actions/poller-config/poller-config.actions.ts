import { ActionWithPayload, PollerConfig } from "@vrbo/data-models";

export enum PollerConfigActions {
  SetNewPollerConfig = "[PollerConfig] Set new poller config"
}

export const setNewPollerConfigAction = (
  newConfig: PollerConfig
): ActionWithPayload<PollerConfig> => ({
  type: PollerConfigActions.SetNewPollerConfig,
  payload: newConfig
});
