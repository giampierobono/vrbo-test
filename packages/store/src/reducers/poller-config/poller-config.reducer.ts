import {
  ActionWithPayload,
  PollerConfig,
  PropertyType
} from "@vrbo/data-models";
import { PollerConfigActions } from "../../actions";

export interface PollerConfigState {
  config: PollerConfig;
}

export const initialPollerConfigState: PollerConfigState = {
  config: {
    limit: 10,
    type: PropertyType.Houses
  }
};

export const pollerConfigReducer = (
  state: PollerConfigState = initialPollerConfigState,
  action: ActionWithPayload<PollerConfig>
) => {
  switch (action.type) {
    case PollerConfigActions.SetNewPollerConfig: {
      return {
        ...state,
        config: {
          ...action.payload
        }
      };
    }
    default: {
      return state;
    }
  }
};
