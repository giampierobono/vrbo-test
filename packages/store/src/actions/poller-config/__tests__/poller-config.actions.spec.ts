import { PropertyType } from "@vrbo/data-models";
import { setNewPollerConfigAction } from "../poller-config.actions";

describe("Poller config actions", () => {
  it("should return correct setNewPollerConfigAction object for type houses", () => {
    expect(
      setNewPollerConfigAction({ type: PropertyType.Houses, limit: 10 })
    ).toMatchSnapshot();
  });

  it("should return correct setNewPollerConfigAction object for type condos", () => {
    expect(
      setNewPollerConfigAction({ type: PropertyType.Condos, limit: 10 })
    ).toMatchSnapshot();
  });
});
