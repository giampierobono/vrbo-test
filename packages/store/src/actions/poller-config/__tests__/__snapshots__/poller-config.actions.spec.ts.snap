// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poller config actions should return correct setNewPollerConfigAction object for type condos 1`] = `
Object {
  "payload": Object {
    "limit": 10,
    "type": "condos",
  },
  "type": "[PollerConfig] Set new poller config",
}
`;

exports[`Poller config actions should return correct setNewPollerConfigAction object for type houses 1`] = `
Object {
  "payload": Object {
    "limit": 10,
    "type": "houses",
  },
  "type": "[PollerConfig] Set new poller config",
}
`;
